spec/unit/typesafe/config/config_value_spec.rb

```diff
@@ -641,3 +647,249 @@ def n(v)
     ["a", "b", "c"].map { |x| expect(falses.get_boolean(x)).to be false }
   end
 end
+
+describe "SimpleConfigOrigin" do
+  let(:has_filename) { SimpleConfigOrigin.new_file("foo") }
+  let(:no_filename) { SimpleConfigOrigin.new_simple("bar") }
+  let(:filename_with_line) { has_filename.with_line_number(3) }
+  let(:no_filename_with_line) { no_filename.with_line_number(4) }
+
+  specify "filename matches what was specified" do
+    expect(has_filename.filename).to eq("foo")
+    expect(filename_with_line.filename).to eq("foo")
+    expect(no_filename.filename).to be nil
+    expect(no_filename_with_line.filename).to be nil
+  end
+
+  specify "description matches correctly" do
+    expect(has_filename.description).to eq("foo")
+    expect(no_filename.description).to eq("bar")
+    expect(filename_with_line.description).to eq("foo: 3")
+    expect(no_filename_with_line.description).to eq("bar: 4")
+  end
+
+  specify "origins with no line number should have line number of -1" do
+    expect(has_filename.line_number).to eq(-1)
+    expect(no_filename.line_number).to eq(-1)
+  end
+
+  specify "line_number returns the right line number" do
+    expect(filename_with_line.line_number).to eq(3)
+    expect(no_filename_with_line.line_number).to eq(4)
+  end
+end
+
+describe "Config#with_only_key and with_only_path" do
+  specify "should keep the correct data" do
+    object = TestUtils.parse_object("{ a=1, b=2, c.d.y=3, e.f.g=4, c.d.z=5 }")
+
+    expect(object.with_only_key("a")).to eq(TestUtils.parse_object("{ a=1 }"))
+    expect(object.with_only_key("e")).to eq(TestUtils.parse_object("{ e.f.g=4 }"))
+    expect(object.to_config.with_only_path("c.d").root).to eq(TestUtils.parse_object("{ c.d.y=3, c.d.z=5 }"))
+    expect(object.to_config.with_only_path("c.d.z").root).to eq(TestUtils.parse_object("{ c.d.z=5 }"))
+
+    expect(object.with_only_key("nope")).to eq(TestUtils.parse_object("{ }"))
+    expect(object.to_config.with_only_path("q.w.e.r.t.y").root).to eq(TestUtils.parse_object("{ }"))
+    expect(object.to_config.with_only_path("a.nonextistent").root).to eq(TestUtils.parse_object("{ }"))
+    expect(object.to_config.with_only_path("c.d.z.nonexistent").root).to eq(TestUtils.parse_object("{ }"))
```

Comment:
I think it would be worth adding `it` blocks around these expectation that contain the messages in the upstream version of the test, eg, `"keep only a"`, etc.